src/System.Runtime.Extensions/src/System/Environment.WinRT.cs

```diff
@@ -0,0 +1,124 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Windows.Foundation.Metadata;
+using Windows.Storage;
+
+namespace System
+{
+    public static partial class Environment
+    {
+        private static string GetFolderPathCore(SpecialFolder folder, SpecialFolderOption option)
+        {
+            // For testing we'll fall back if the needed APIs aren't present
+            if (ApiInformation.IsTypePresent("Windows.Storage.UserDataPaths"))
+            {
+                return GetFolderPathCoreCurrent(folder, option);
+            }
+            else
+            {
+                return GetFolderPathCoreFallBack(folder, option);
+            }
+
+        }
+
+        private static string GetFolderPathCoreCurrent(SpecialFolder folder, SpecialFolderOption option)
```

Comment:
I'm not understanding the rationale.  Our goal here is to achieve as much fidelity as possible with the existing .NET API, not with the underlying API we build on; if the underlying API doesn't provide the functionality but we can achieve it in other ways, why shouldn't we?  The default behavior is to verify that the path being handed back actually exists, and if it doesn't, give back an empty string so that consuming code doesn't try to use something that doesn't exist; by not doing that, it seems like we're artificially harming consuming code. Is there some reason we can't simply do the exists check?  As for Create, I understand the concerns, but code has to explicitly opt-in by passing in Create; if they do so, shouldn't we respect that?  If we don't, a) existing code that expected it won't work correctly and b) such code would then potentially be modified to work around it by doing the exact creation option we'd attempt.